scheduler/filter/utils.go

```diff
@@ -0,0 +1,89 @@
+package filter
+
+import (
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type comparison int
+
+const (
+	equ = comparison(iota)
+	neg
+	gte
+	lte
+)
+
+func parse(k, v string) (string, string, comparison, bool) {
+	// default comparison mode
+	mode := equ
+
+	// support case of constraint:k==v
+	// try to check = on both sides, "k= : = : v" and "k : = : =v", to make sure it works everywhere
+	if strings.HasSuffix(k, "=") {
+		k = strings.TrimSuffix(k, "=")
+	} else if strings.HasPrefix(v, "=") {
+		v = strings.TrimPrefix(v, "=")
+	}
+
+	if strings.HasPrefix(v, "!") {
```

Comment:
I personally think this will become harder to maintain when we add more operators and more cases to handle.
How about instead of splitting the constraint on "=" and then trying to find each operator in either key/val, you define all the operators (ie: "=", "==", ">=", "<=", "!=", "=!", and each operator has a matching mode.
Then all you have to do something like the following:

func findMode(constraint string) {
for(op <- operators) {
 if strings.Contains(constraints, op.value) {
    return op.mode
 }
}

 err....
}